legacy/c3_trajectory_generator/src/node.cpp

```diff
@@ -191,12 +235,34 @@ struct Node {
       c3trajectory_t = now;
     }
 
-    while (c3trajectory_t + traj_dt < now) {
-      c3trajectory->update(traj_dt.toSec(), current_waypoint,
-                           (c3trajectory_t - current_waypoint_t).toSec());
+    // Remember the previous trajectory
+    auto old_trajectory = c3trajectory->getCurrentPoint();
+
+    while (c3trajectory_t + traj_dt < now)
+    {
+      c3trajectory->update(traj_dt.toSec(), current_waypoint, (c3trajectory_t - current_waypoint_t).toSec());
       c3trajectory_t += traj_dt;
     }
 
+    // Check if we will hit something while in trajectory the new trajectory
+    geometry_msgs::Pose traj_point;  // Convert messages to correct type
+    auto p = c3trajectory->getCurrentPoint();
+    traj_point.position = vec2xyz<Point>(p.q.head(3));
+    quaternionTFToMsg(tf::createQuaternionFromRPY(p.q[3], p.q[4], p.q[5]), traj_point.orientation);
+
+    std::pair<bool, WAYPOINT_ERROR_TYPE> checkWPResult =
+        waypoint_validity_.is_waypoint_valid(Pose_from_Waypoint(current_waypoint));
+    if (checkWPResult.first == false && checkWPResult.second == WAYPOINT_ERROR_TYPE::OCCUPIED)
```

Comment:
I see what you mean, although I'm not sure we want to proceed quietly when we are navigating without occupation information.